src/validation.cpp

```diff
@@ -3032,6 +3040,13 @@ bool Chainstate::ActivateBestChain(BlockValidationState& state, std::shared_ptr<
     // we use m_chainstate_mutex to enforce mutual exclusion so that only one caller may execute this function at a time
     LOCK(m_chainstate_mutex);
 
+    // Belt-and-suspenders check that we aren't attempting to advance the background
+    // chainstate past the snapshot base block.
```

Comment:
From my understanding there is a marginal case where this is not *only* a belt-and-suspender. When background chainstate reaches height parity with the snapshot chainstate, the background chainstate is torn down only in `LoadChainstate->ValidatedSnapshotCleanup` by the call to `ResetChainstates()` L5584. Before, ABC could be called as the background chainstate is still `ActiveChainstate()`, at least for the block of the connection sequence, before the remaining one hit the `m_disabled` L3106. If correct, the comment could be more informative.

Reply:
For what it's worth, in future PRs I think this possibility is eliminated, since calls like `ActiveChainstate().ActivateBestChain()` in ProcessNewBlock are replaced with specific chainstate references.